appinterface/projection/rdbprojectionbase/base.go

```diff
@@ -57,10 +58,14 @@ func NewRDbBaseWithOptions(rdbHandle *rdb.Handle, projectionId string, options O
 }
 
 type Options struct {
-	// Pointer to the configuration
+	// Pointer to the configuration to be read from Toml file
+	MaybeReadConfigPtr interface{}
```

Comment:
our app should not specifically binding to Toml config file, may i know why we need `MaybeReadConfigPtr`?

Reply:
Same question here. After adding this, it seems `BridgeActivityMatcher.New()` will not read config from toml file.

Reply:
The original behaviour of `MaybeConfigPtr` is to read from the global projection config which is reading from Toml file:
https://github.com/crypto-com/chain-indexing/pull/640/files#diff-2bf44a05f77f5d8ea19f90ed17e81e60731cff8b12fa4d096f98b61b3088bfe4R42

`GlobalConfig` is a singleton https://github.com/crypto-com/chain-indexing/blob/master/usecase/projection/config.go
read in app

Reply:
For the test case I found it easier to inject the config directly is more readable, that's why I introduce a new way to provide the config.

In the process I found the original `MaybeConfigPtr` is also confusion, and hence I update the naming as well.

Reply:
But I agree the assumption of it always being read from Toml may not hold in the future.

Reply:
Opened a new issue to handle this issue. https://github.com/crypto-com/chain-indexing/issues/648

Will put this PR on hold and work on it first.